transmogrifier/sources/json/mitlibwebsite.py

```diff
@@ -0,0 +1,83 @@
+import hashlib
+import logging
+
+import transmogrifier.models as timdex
+from transmogrifier.sources.jsontransformer import JSONTransformer
+from transmogrifier.sources.transformer import JSON
+
+logger = logging.getLogger(__name__)
+
+
+class MITLibWebsite(JSONTransformer):
+
+    @classmethod
+    def get_main_titles(cls, source_record: dict) -> list[str]:
+        """
+        Retrieve main title(s) from a JSON record.
+
+        Must be overridden by source subclasses.
+
+        Args:
+            source_record: A JSON object representing a source record.
+        """
+        if source_record["cdx_title"] is None:
+            return []
+        return [source_record["cdx_title"]]
+
+    def get_source_link(self, source_record: dict) -> str:
+        """Set source link for the item.
+
+        Unlike other Transmogrifier sources that dynamically build a source link,
+        MITLibWebsite files are expected to have a fully formed and appropriate
+        source link in the metadata already.  This method relies on that data.
+        """
+        return source_record["url"]
+
+    def get_timdex_record_id(self, source_record: dict) -> str:
+        """Set the TIMDEX record ID.
+
+        Concatenate the source name + source record id.
+        """
+        return f"{self.source}:{self.get_source_record_id(source_record)}"
+
+    @classmethod
+    def get_source_record_id(cls, source_record: dict) -> str:
+        """Get or generate a source record ID from a JSON record.
+
+        Generates an MD5 hash of the 'url' value.
```

Comment:
_Late request so no worries if it doesn't make it into a commit in this PR..._

Wondering if maybe we can add a bit more to this docstring?  I think it could be helpful to explain:

- the website URL is the most unique and consistent identifier for this record, making it a great identifier
- _but_, it makes a poor `timdex_record_id` given the special characters and long nature
- hence the MD5 hash of the URL

Reply:
Thanks for the suggestion!